mods/default/functions.lua

```diff
@@ -293,13 +293,41 @@ minetest.register_abm({
 -- Dig upwards
 --
 
-function default.dig_up(pos, node, digger)
+local in_dig_up = false
+
+---Find all nodes above this one that is the same, then dig them all
+---@param pos vector The position of the base node
+---@param node { name: string, param1: integer, param2: integer } Node table of the base node
+---@param digger ObjectRef The object (e.g. player) digging the node
+---@param max_height The maximum height to search for, excluding the base node
+function default.dig_up(pos, node, digger, max_height)
+	if in_dig_up then return end -- Do not recurse
 	if digger == nil then return end
-	local np = {x = pos.x, y = pos.y + 1, z = pos.z}
-	local nn = minetest.get_node(np)
-	if nn.name == node.name then
-		minetest.node_dig(np, nn, digger)
+	max_height = max_height or 100
+
+	in_dig_up = true
+	for y = pos.y + 1, pos.y + max_height do
+		local up_pos  = vector.new(pos.x, y, pos.z)
+		local up_node = minetest.get_node(up_pos)
+		if up_node.name ~= node.name then
+			break
+		end
+		local noerr, success = xpcall(function()
```

Comment:
thinking about this again: why does it even matter that it's left in a broken state when an error happens?
i don't see any situation in which mods doing a try-catch with dig_node makes sense

Reply:
I don't think that whether it makes sense actually limits the creativity of mod authors. It's an edge-case for sure.

Reply:
Technically there are *still* ways to force a broken state if you want to, e.g. using coroutines. But I don't think it's very worthwhile to try to guard against this.

Reply:
additional idea: reset `in_dig_up` on every server step

this minimizes the effects of the already unlikely error edge case to basically nothing

Reply:
Done in b49153c